src/OpenColorIO/ops/Gamma/GammaOpCPU.cpp

```diff
@@ -0,0 +1,439 @@
+                                 std::max(0.0f, rgba[2]),
+                                 std::max(0.0f, rgba[3]) };
+
+        rgba[0] = powf(pixel[0] * m_inScale, m_redGamma) * m_outScale;
+        rgba[1] = powf(pixel[1] * m_inScale, m_grnGamma) * m_outScale;
+        rgba[2] = powf(pixel[2] * m_inScale, m_bluGamma) * m_outScale;
+        rgba[3] = powf(pixel[3] * m_inScale, m_alpGamma) * m_outScale;
+
+        rgba += 4;
+    }
+#endif
+}
+
+GammaMoncurveOpCPUFwd::GammaMoncurveOpCPUFwd(const GammaOpDataRcPtr & gamma)
+    :   GammaMoncurveOpCPU(gamma)
+    ,   m_outScale(0.0f)
+{
+    update(gamma);
+}
+
+void GammaMoncurveOpCPUFwd::update(const GammaOpDataRcPtr & gamma)
+{
+    // NB: The power function is applied in normalized space
+    // but we fold the in/out depth conversion into the other scaling
+    // to minimize the number of multiplies.
+
+    const BitDepth inBitDepth  = gamma->getInputBitDepth();
+    const BitDepth outBitDepth = gamma->getOutputBitDepth();
+
+    ComputeParamsFwd(gamma->getRedParams(),   inBitDepth, outBitDepth, m_red);
+    ComputeParamsFwd(gamma->getGreenParams(), inBitDepth, outBitDepth, m_green);
+    ComputeParamsFwd(gamma->getBlueParams(),  inBitDepth, outBitDepth, m_blue);
+    ComputeParamsFwd(gamma->getAlphaParams(), inBitDepth, outBitDepth, m_alpha);
+
+    m_outScale = GetBitDepthMaxValue(outBitDepth);
+}
+
+void GammaMoncurveOpCPUFwd::apply(float * rgbaBuffer, long numPixels) const
+{
+#ifdef USE_SSE
+    const __m128 scale
+      = _mm_set_ps((float)m_alpha.scale, (float)m_blue.scale,
+                   (float)m_green.scale, (float)m_red.scale);
+
+    const __m128 offset
+      = _mm_set_ps((float)m_alpha.offset, (float)m_blue.offset,
+                   (float)m_green.offset, (float)m_red.offset);
+
+    const __m128 gamma
+      = _mm_set_ps((float)m_alpha.gamma, (float)m_blue.gamma,
+                   (float)m_green.gamma, (float)m_red.gamma);
+
+    const __m128 breakPnt
+      = _mm_set_ps((float)m_alpha.breakPnt, (float)m_blue.breakPnt,
+                   (float)m_green.breakPnt, (float)m_red.breakPnt);
+
+    const __m128 slope
+      = _mm_set_ps((float)m_alpha.slope, (float)m_blue.slope,
+                   (float)m_green.slope, (float)m_red.slope);
+
+    const __m128 outScale = _mm_set1_ps(m_outScale);
+
+    float * rgba = rgbaBuffer;
+
+    for(long idx=0; idx<numPixels; ++idx)
+    {
+        __m128 pixel = _mm_set_ps(rgba[3], rgba[2], rgba[1], rgba[0]);
+
+        __m128 data = _mm_add_ps(_mm_mul_ps(pixel, scale), offset);
+
+        data = ssePower(data, gamma);
+
+        data = _mm_mul_ps(data, outScale);
+
+        __m128 flag = _mm_cmpgt_ps( pixel, breakPnt);
+
+        data = _mm_or_ps(_mm_and_ps(flag, data),
+                         _mm_andnot_ps(flag, _mm_mul_ps(pixel, slope )));
+
+        _mm_storeu_ps(rgba, data);
+
+        rgba += 4;
+    }
+#else
+    float * rgba = rgbaBuffer;
+
+    const float red[5] 
+        = { float(m_red.scale),  float(m_red.offset),

```

Comment:
Is there a reason for carrying the `double` render parameters everywhere if they're just going to cast to `float`?

Reply:
We are slowly transitioning the Ops (i.e. OpData) to double so that the GPU implementation benefits from the double precision, which imposes the RenderParams structure to handle double values. But the CPU implementations (i.e. SEE and not SEE) still computes from float values (i.e. the pixel) needing a cast of the Gamma parameters outside the for-loop processing. As the CPU performance is critical, the transition to a double precision computation within the for-loop should be investigated first.

Reply:
Patrick and I discussed this and he will look at adding a commit for this.

Reply:
Done